docs/configuration.md

```diff
@@ -1141,6 +1141,48 @@ Apart from these, the following properties are also available, and may be useful
   </td>
 </tr>
 <tr>
+  <td><code>spark.scheduler.blacklist.enabled</code></td>
+  <td>true</td>
+  <td>
+    If set to ture, executor blacklist feature will be enabled to avoid allocte new task on bad executor. The logic to define bad executor is based on BlacklistStrategy which is also configurable.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.scheduler.executorTaskBlacklistTime</code></td>
+  <td>0L</td>
+  <td>
+    The threshold to deside blacklist executor, if last failed time is older than current time subtract executorTaskBlacklistTime, then the executor could be removed from blacklist.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.scheduler.blacklist.recoverPeriod</code></td>
+  <td>60L</td>
+  <td>
+    The period between runnning blacklist recover process. 
+  </td>
+</tr>
+<tr>
+  <td><code>spark.scheduler.blacklist.strategy</code></td>
+  <td>default</td>
+  <td>
+    The strategy to determine executor blacklist and node blacklist. There are "default" strategy which is the srandard behavior before spark 1.6 and "threshold" strategy which is taskId unrelated and can learn experience from other taskSet to avoid allocating tasks on problematic executors.
```

Comment:
this definitely needs more thorough documentation ... probably more than should go into the description of a parameter. Maybe it deserves an additional section like http://spark.apache.org/docs/latest/job-scheduling.html#dynamic-resource-allocation ?
I will think about this a bit more ...